### PR TITLE
Winter 23 api update and update test classes to use new Assert class

### PR DESCRIPTION
**Pull request by chazwatkins**
This change is [Reviewable](https://reviewable.io/reviews/apex-enterprise-patterns/at4dx/68)

### CODE REVIEW

File: sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls [hunk 1]
```diff
@@ -114,8 +114,8 @@ public class PlatformEventDistributorTest
         // Inject the platform event directly into the trigger handler, bypassing event publishing
         new PlatformEventDistributor().triggerHandler(new List<AT4DXMessage__e> { platformEventMessage });
 
-        System.assertEquals(true, matchEventBusAndCategoryResult.wasTestPlatformEventsConsumerCalled, 'Platform events consumer was not called');
-        System.assertEquals('Test Payload', matchEventBusAndCategoryResult.testPlatformEventsConsumerData, 'Payload to consumer did not match');        
+        System.Assert.areEqual(true, matchEventBusAndCategoryResult.wasTestPlatformEventsConsumerCalled, 'Platform events consumer was not called');
```
Comment (ImJohnMDaniel) [inline sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls:117]: Please change to `System.Assert.isTrue()` method

File: sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls [hunk 2]
```diff
@@ -155,8 +155,8 @@ public class PlatformEventDistributorTest
         // Inject the platform event directly into the trigger handler, bypassing event publishing
         new PlatformEventDistributor().triggerHandler(new List<AT4DXMessage__e> { platformEventMessage });
 
-        System.assertEquals(true, matchEventBusAndEventNameResult.wasTestPlatformEventsConsumerCalled, 'Platform events consumer was not called');
-        System.assertEquals('Test Payload', matchEventBusAndEventNameResult.testPlatformEventsConsumerData, 'Payload to consumer did not match');    
+        System.Assert.areEqual(true, matchEventBusAndEventNameResult.wasTestPlatformEventsConsumerCalled, 'Platform events consumer was not called');
```
Comment (ImJohnMDaniel) [inline sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls:158]: Please change to `System.Assert.isTrue()` method

File: sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls [hunk 3]
```diff
@@ -196,8 +196,8 @@ public class PlatformEventDistributorTest
         // Inject the platform event directly into the trigger handler, bypassing event publishing
         new PlatformEventDistributor().triggerHandler(new List<AT4DXMessage__e> { platformEventMessage });
 
-        System.assertEquals(true, matchEventBusResult.wasTestPlatformEventsConsumerCalled, 'Platform events consumer was not called');
-        System.assertEquals('Test Payload', matchEventBusResult.testPlatformEventsConsumerData, 'Payload to consumer did not match');    
+        System.Assert.areEqual(true, matchEventBusResult.wasTestPlatformEventsConsumerCalled, 'Platform events consumer was not called');
```
Comment (ImJohnMDaniel) [inline sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls:199]: Please change to `System.Assert.isTrue()` method

File: sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls [hunk 4]
```diff
@@ -238,7 +238,7 @@ public class PlatformEventDistributorTest
         PlatformEventDistributor pedInstance = new PlatformEventDistributor();
         pedInstance.triggerHandler(new List<AT4DXMessage__e> { platformEventMessage });
 
-        System.assertEquals(false, pedInstance.jobIdList.isEmpty(), 'Job id was not found in the job id list');  
+        System.Assert.areEqual(false, pedInstance.jobIdList.isEmpty(), 'Job id was not found in the job id list');
```
Comment (ImJohnMDaniel) [inline sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls:241]: Please change to `System.Assert.isFalse()` method

**Review by ImJohnMDaniel — CHANGES_REQUESTED** [review]

**Comment by chazwatkins** [comment]
@ImJohnMDaniel  Good catches.  With so many changes, I figured there would be a few left.

**Comment by chazwatkins** [comment]
The updates are finished as requested

File: sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls [hunk 5]
```diff
@@ -114,8 +114,8 @@ public class PlatformEventDistributorTest
         // Inject the platform event directly into the trigger handler, bypassing event publishing
         new PlatformEventDistributor().triggerHandler(new List<AT4DXMessage__e> { platformEventMessage });
 
-        System.assertEquals(true, matchEventBusAndCategoryResult.wasTestPlatformEventsConsumerCalled, 'Platform events consumer was not called');
-        System.assertEquals('Test Payload', matchEventBusAndCategoryResult.testPlatformEventsConsumerData, 'Payload to consumer did not match');        
+        System.Assert.isTruematchEventBusAndCategoryResult.wasTestPlatformEventsConsumerCalled, 'Platform events consumer was not called');
```
Comment (ImJohnMDaniel) [inline sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls:117]: @chazwatkins -- minor syntax correction needed here please.

File: sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls [hunk 6]
```diff
@@ -155,8 +155,8 @@ public class PlatformEventDistributorTest
         // Inject the platform event directly into the trigger handler, bypassing event publishing
         new PlatformEventDistributor().triggerHandler(new List<AT4DXMessage__e> { platformEventMessage });
 
-        System.assertEquals(true, matchEventBusAndEventNameResult.wasTestPlatformEventsConsumerCalled, 'Platform events consumer was not called');
-        System.assertEquals('Test Payload', matchEventBusAndEventNameResult.testPlatformEventsConsumerData, 'Payload to consumer did not match');    
+        System.Assert.isTruematchEventBusAndEventNameResult.wasTestPlatformEventsConsumerCalled, 'Platform events consumer was not called');
```
Comment (ImJohnMDaniel) [inline sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls:158]: same here

File: sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls [hunk 7]
```diff
@@ -196,8 +196,8 @@ public class PlatformEventDistributorTest
         // Inject the platform event directly into the trigger handler, bypassing event publishing
         new PlatformEventDistributor().triggerHandler(new List<AT4DXMessage__e> { platformEventMessage });
 
-        System.assertEquals(true, matchEventBusResult.wasTestPlatformEventsConsumerCalled, 'Platform events consumer was not called');
-        System.assertEquals('Test Payload', matchEventBusResult.testPlatformEventsConsumerData, 'Payload to consumer did not match');    
+        System.Assert.isTruematchEventBusResult.wasTestPlatformEventsConsumerCalled, 'Platform events consumer was not called');
```
Comment (ImJohnMDaniel) [inline sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls:199]: same here

File: sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls [hunk 8]
```diff
@@ -238,7 +238,7 @@ public class PlatformEventDistributorTest
         PlatformEventDistributor pedInstance = new PlatformEventDistributor();
         pedInstance.triggerHandler(new List<AT4DXMessage__e> { platformEventMessage });
 
-        System.assertEquals(false, pedInstance.jobIdList.isEmpty(), 'Job id was not found in the job id list');  
+        System.Assert.isTruepedInstance.jobIdList.isEmpty(), 'Job id was not found in the job id list');
```
Comment (ImJohnMDaniel) [inline sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls:241]: same here

**Review by ImJohnMDaniel — CHANGES_REQUESTED** [review]

File: sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls (hunk 5 above)
Comment (chazwatkins) [reply sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls:117]: Oof, don't know how I managed that one.  I'll update it now

**Comment by chazwatkins** [comment]
Apologies for the typos.  The changes are now complete

**Review by ImJohnMDaniel — APPROVED** [review]

**Review by stohn777 — APPROVED** [review]

**Review by chazwatkins — COMMENTED** [review]
Resolving all the reviewable issues.  I believe all the requested changes are complete.

> Reviewed 53 of 55 files at r1, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/apex-enterprise-patterns/at4dx/68#-:-NLwYyYpDmekJ0cCD5zL:b-pquv0n)* status: all files reviewed, 8 unresolved discussions (waiting on @ImJohnMDaniel)

___
*[`sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls` line 117 at r1](https://reviewable.io/reviews/apex-enterprise-patterns/at4dx/68#-NIJcT8jjOWTnpfgOP-1-r1-117:-NLwY-2v7jofZdB5UvIa:b-896fix) ([raw file](https://github.com/apex-enterprise-patterns/at4dx/blob/5aa96e26d6771ac0e309d91b8cee0bd27e08c4e1/sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls#L117)):*
<details><summary><i>Previously, ImJohnMDaniel (John M. Daniel) wrote…</i></summary><blockquote>

Please change to `System.Assert.isTrue()` method
</blockquote></details>

Done.
___
*[`sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls` line 158 at r1](https://reviewable.io/reviews/apex-enterprise-patterns/at4dx/68#-NIJcT8jjOWTnpfgOP-1-r1-158:-NLwY7ql5moyuYeognp3:b-896fix) ([raw file](https://github.com/apex-enterprise-patterns/at4dx/blob/5aa96e26d6771ac0e309d91b8cee0bd27e08c4e1/sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls#L158)):*
<details><summary><i>Previously, ImJohnMDaniel (John M. Daniel) wrote…</i></summary><blockquote>

Please change to `System.Assert.isTrue()` method
</blockquote></details>

Done.
___
*[`sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls` line 199 at r1](https://reviewable.io/reviews/apex-enterprise-patterns/at4dx/68#-NIJcT8jjOWTnpfgOP-1-r1-199:-NLwYuNx6g97yyh0kl0P:b-896fix) ([raw file](https://github.com/apex-enterprise-patterns/at4dx/blob/5aa96e26d6771ac0e309d91b8cee0bd27e08c4e1/sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls#L199)):*
<details><summary><i>Previously, ImJohnMDaniel (John M. Daniel) wrote…</i></summary><blockquote>

Please change to `System.Assert.isTrue()` method
</blockquote></details>

Done.
___
*[`sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls` line 241 at r1](https://reviewable.io/reviews/apex-enterprise-patterns/at4dx/68#-NIJcT8jjOWTnpfgOP-1-r1-241:-NLwYw6x5IC5exdSKa0m:b-896fix) ([raw file](https://github.com/apex-enterprise-patterns/at4dx/blob/5aa96e26d6771ac0e309d91b8cee0bd27e08c4e1/sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls#L241)):*
<details><summary><i>Previously, ImJohnMDaniel (John M. Daniel) wrote…</i></summary><blockquote>

Please change to `System.Assert.isFalse()` method
</blockquote></details>

Done.
___
*[`sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls` line 158 at r2](https://reviewable.io/reviews/apex-enterprise-patterns/at4dx/68#-NIJcT8jjOWTnpfgOP-1-r2-158:-NLwYsjTDjRimGoYCh-x:b-896fix) ([raw file](https://github.com/apex-enterprise-patterns/at4dx/blob/4777ff4f1efe9fe8e7587b436261b5d6d6808675/sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls#L158)):*
<details><summary><i>Previously, ImJohnMDaniel (John M. Daniel) wrote…</i></summary><blockquote>

same here
</blockquote></details>

Done.
___
*[`sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls` line 199 at r2](https://reviewable.io/reviews/apex-enterprise-patterns/at4dx/68#-NIJcT8jjOWTnpfgOP-1-r2-199:-NLwYv1xA1VLwKgxB5ZK:b-896fix) ([raw file](https://github.com/apex-enterprise-patterns/at4dx/blob/4777ff4f1efe9fe8e7587b436261b5d6d6808675/sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls#L199)):*
<details><summary><i>Previously, ImJohnMDaniel (John M. Daniel) wrote…</i></summary><blockquote>

same here
</blockquote></details>

Done.
___
*[`sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls` line 241 at r2](https://reviewable.io/reviews/apex-enterprise-patterns/at4dx/68#-NIJcT8jjOWTnpfgOP-1-r2-241:-NLwYwTY3MOskkz9lZb-:b-896fix) ([raw file](https://github.com/apex-enterprise-patterns/at4dx/blob/4777ff4f1efe9fe8e7587b436261b5d6d6808675/sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls#L241)):*
<details><summary><i>Previously, ImJohnMDaniel (John M. Daniel) wrote…</i></summary><blockquote>

same here
</blockquote></details>

Done.

**Comment by chazwatkins** [comment]
*[`sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls` line 117 at r2](https://reviewable.io/reviews/apex-enterprise-patterns/at4dx/68#-NIJcT8jjOWTnpfgOP-1-r2-117:-NLwZDso5OOTa0qAazCv:b-896fix) ([raw file](https://github.com/apex-enterprise-patterns/at4dx/blob/4777ff4f1efe9fe8e7587b436261b5d6d6808675/sfdx-source/core/test/classes/framework-platform-event-distributor/PlatformEventDistributorTest.cls#L117)):*
<details><summary><i>Previously, chazwatkins (Chaz Watkins) wrote…</i></summary><blockquote>

Oof, don't know how I managed that one.  I'll update it now
</blockquote></details>

Done.

**Review by chazwatkins — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/apex-enterprise-patterns/at4dx/68#-:-NLwZNn_-_3ToXN0CZCj:bnfp4nl)* status: all files reviewed, 8 unresolved discussions (waiting on @ImJohnMDaniel)

**Comment by ImJohnMDaniel** [comment]
@chazwatkins -- Thanks for the PR.  

Reviewable is interesting, for certain.  Not required on the AEP repos but some of us do us it.

**Pull request merged**